src/com/service/CreatePersonService.java

```diff
@@ -0,0 +1,178 @@
+    public static final String STATUS_INPUT = "Enter your status *** 0 for single <-> 1 for in relationship <-> 2 for married";
+    public static final String INVALID_INPUT = "Invalid %s input";
+    public static final String SUCCESSFUL_INPUT = "%s successfully added";
+    public static final String INTEREST_CHAR_LIMIT = "Your interests have been reduced to 250 characters";
+    public static final String NAME_VALIDATION_REGEX = "[a-zA-z]{3,}";
+    public static final String USE_INTEGERS_ONLY = "Use integers as per description";
+    public static final String PATTERN_DATE_OF_BIRTH = "yyyy-MM-dd";
+
+    private static final Scanner scannerIn = new Scanner(System.in);
+
+    public static int getUserInputInteger() {
+        return Integer.parseInt(scannerIn.nextLine());
+    }
+
+    public static String getUserInputString() {
+        return scannerIn.nextLine();
+    }
+
+    public static String wrongInputByUserReturnNewLine() {
+        return scannerIn.nextLine();
+    }
+
+    public static boolean scannerHasNextLine() {
+        return scannerIn.hasNextLine();
+    }
+
+    public static boolean scannerHasNextInt() {
+        return scannerIn.hasNextInt();
+    }
+
+    public String createPersonName() {
+        while (true) {
+            System.out.println(NAME_INPUT);
+            if (scannerHasNextLine()) {
+                String name = getUserInputString();
+                if (name != null && name.matches(NAME_VALIDATION_REGEX)) {
+                    System.out.format(SUCCESSFUL_INPUT, name);
+                    System.out.println();
+                    return name;
+                }
+            } else {
+                wrongInputByUserReturnNewLine();
+                System.out.println(INVALID_INPUT);
+            }
+        }
+    }
+
+    public EGender createPersonGender() {
+        System.out.println(GENDER_INPUT);
+        EGender gender;
+        while (true) {
+            if (scannerHasNextInt()) {
+                try {
+                    int genderInput = getUserInputInteger();
+                    if (genderInput >= 0 && genderInput <= 1) {
+                        gender = EGender.values()[genderInput];
+                        System.out.format(SUCCESSFUL_INPUT, gender);
+                        System.out.println();
+                        return gender;
+                    } else if (genderInput > 1) {
+                        wrongInputByUserReturnNewLine();
+                        System.out.println(USE_INTEGERS_ONLY);
+                    }
+                } catch (NumberFormatException e) {
+                    wrongInputByUserReturnNewLine();
+                    System.out.println(USE_INTEGERS_ONLY + e.getMessage());
+                }
+            } else {
+                wrongInputByUserReturnNewLine();
+                System.out.println(USE_INTEGERS_ONLY);
+            }
+        }
+    }
+
+    public static Date formatDateOfBirth(String dateOfBirthString) {
```

Comment:
This should be in a separate class named utils. I will explain this

Reply:
Sorted